metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1349,7 +1349,7 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode[SourceDataSetT
                     col_ref=SqlColumnReference(table_alias=parent_alias, column_name=metric_time_dimension_column_name)
                 ),
             ),
-            join_type=SqlJoinType.LEFT_OUTER,  # TODO: test other join types
+            join_type=SqlJoinType.INNER,
```

Comment:
shall we move the # Build join expression block to sql_join_builder.py? Sorry I missed this before, but Tom made some efforts to clean this stuff up by splitting out building join logics to that module.

Reply:
Sure!